### PR TITLE
ref(slack): Convert the text portion of issue alerts to use block kit

### PR DESCRIPTION
**Pull request by ceorourke**
Breaking up https://github.com/getsentry/sentry/pull/61215 into smaller chunks - this PR is only the text portion of the issue alert (with no buttons). 

Here is the issue alert (note no buttons) built with block kit
Screenshot 2023-12-14 at 12 18 50 PM

### CODE REVIEW

File: src/sentry/features/__init__.py [hunk 1]
```diff
@@ -225,6 +225,7 @@
 default_manager.add("organizations:session-replay-enable-canvas", OrganizationFeature, FeatureHandlerStrategy.REMOTE)
 default_manager.add("organizations:set-grouping-config", OrganizationFeature, FeatureHandlerStrategy.INTERNAL)
 default_manager.add("organizations:slack-overage-notifications", OrganizationFeature, FeatureHandlerStrategy.REMOTE)
+default_manager.add("organizations:slack-block-kit", OrganizationFeature, FeatureHandlerStrategy.REMOTE)
```
Comment (isabellaenriquez) [inline src/sentry/features/__init__.py:228]: we're using flagr to rollout?

File: src/sentry/integrations/slack/message_builder/base/block.py [hunk 2]
```diff
@@ -58,10 +94,30 @@ class SlackBlockType(TypedDict):
 
         return action_block
 
+    @staticmethod
+    def get_context_block(text: str, timestamp: Optional[datetime] = None) -> SlackBlock:
+        if timestamp:
+            time = "<!date^{:.0f}^ {} at {} | Sentry Issue>".format(
+                to_timestamp(timestamp), "{date_pretty}", "{time}"
```
Comment (isabellaenriquez) [inline src/sentry/integrations/slack/message_builder/base/block.py:101]:
```suggestion
            time = "<!date^{:.0f}^{} at {} | Sentry Issue>".format(
                to_timestamp(timestamp), "{date_pretty}", "{time}"
```
i believe the removal of the single space capitalizes lol

File: src/sentry/integrations/slack/message_builder/issues.py [hunk 3]
```diff
@@ -285,26 +308,68 @@ def build(self, notification_uuid: str | None = None) -> SlackBody:
...
+        # build tags block
+        if tags:
+            blocks.append(self.get_tags_block(tags))
+
+        # build footer block
+        footer = (
+            self.notification.build_notification_footer(self.recipient, ExternalProviders.SLACK)
+            if self.notification and self.recipient
+            else build_footer(self.group, project, self.rules, SLACK_URL_FORMAT)
+        )
```
Comment (isabellaenriquez) [inline src/sentry/integrations/slack/message_builder/issues.py:362]: could we use the existing `footer` on L292?

**Review by isabellaenriquez — COMMENTED** [review]
nits + make the failing testcase ✨ pretty ✨, looking good overall 🫡

File: src/sentry/features/__init__.py (hunk 1 above)
Comment (ceorourke) [reply src/sentry/features/__init__.py:228]: probably better to use the internal options thing, good callout

**Review by isabellaenriquez — APPROVED** [review]

**Pull request merged**